presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3098,11 +3098,12 @@ private OperatorFactory createHashAggregationOperatorFactory(
         {
             List<VariableReferenceExpression> aggregationOutputVariables = new ArrayList<>();
             List<AccumulatorFactory> accumulatorFactories = new ArrayList<>();
+            boolean useSpill = spillEnabled && !isStreamable && (!hasDistinct(aggregations) || distinctSpillEnabled) && (!hasOrderBy(aggregations) || orderBySpillEnabled);
```

Comment:
Maybe rename `distinctSpillEnabled` to `distinctAggregateSpillEnabled` and `orderBySpillEnabled` to `orderByAggregateSpillEnabled` to make it more clear.